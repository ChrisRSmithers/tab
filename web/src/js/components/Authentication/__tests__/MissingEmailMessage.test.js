/* eslint-env jest */

import React from 'react'
import {
  mount,
  shallow
} from 'enzyme'
import toJson from 'enzyme-to-json'
import {
  logout
} from 'js/authentication/user'
import {
  goTo,
  loginURL
} from 'js/navigation/navigation'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

jest.mock('js/authentication/user')
jest.mock('js/navigation/navigation')

afterEach(() => {
  jest.clearAllMocks()
})

describe('MissingEmailMessage tests', function () {
  it('renders without error', function () {
    const MissingEmailMessage = require('js/components/Authentication/MissingEmailMessage').default
    shallow(
      <MissingEmailMessage />
    )
  })

  it('restarts auth flow when clicking button', done => {
    const MissingEmailMessage = require('js/components/Authentication/MissingEmailMessage').default

    // @material-ui-1-todo: remove MuiThemeProvider wrapper
    const wrapper = mount(
      <MuiThemeProvider>
        <MissingEmailMessage />
      </MuiThemeProvider>
    )

    // @material-ui-1-todo: use specific selector
    const button = wrapper.find('[data-test-id="missing-email-message-button-container"] button')
    button.simulate('click')

    // Dealing with async methods triggered by `simulate`:
    // https://stackoverflow.com/a/43855794/1332513
    setImmediate(() => {
      expect(logout).toHaveBeenCalled()
      expect(goTo).toHaveBeenCalledWith(loginURL)
      done()
    })
  })

  it('matches expected snapshot', function () {
    const MissingEmailMessage = require('js/components/Authentication/MissingEmailMessage').default
    const wrapper = shallow(
      <MissingEmailMessage />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
