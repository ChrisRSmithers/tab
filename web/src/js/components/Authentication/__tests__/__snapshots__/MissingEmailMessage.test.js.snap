// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissingEmailMessage tests matches expected snapshot 1`] = `
<Paper
  circle={false}
  rounded={true}
  style={
    Object {
      "backgroundColor": "#FFF",
      "maxWidth": 400,
      "padding": 24,
    }
  }
  transitionEnabled={true}
  zDepth={1}
>
  <h3>
    We couldn't sign you in
  </h3>
  <p>
    This can happen when Facebook or Google doesn't have your email address on file, which we use to create your account. Please try signing in with another method, such as an email and password.
  </p>
  <span
    data-test-id="missing-email-message-button-container"
    style={
      Object {
        "display": "flex",
        "justifyContent": "flex-end",
        "marginTop": 24,
      }
    }
  >
    <RaisedButton
      disabled={false}
      fullWidth={false}
      label="SIGN IN USING ANOTHER METHOD"
      labelPosition="after"
      onClick={[Function]}
      primary={true}
      secondary={false}
    />
  </span>
</Paper>
`;
